Bug: ValidationError for EventBridgeModel

### Expected Behaviour

When running a lambda from an EventBridge scheduler, EventBridgeModel should be able to parse the event.

### Current Behaviour

EventBridgeModel fails to parse EventBridge scheduler event because of the field detail.

For some reason, the event returned by AWS has `detail` property as a string and not as a dict. I am wondering if it's because it is empty.

Exemple of the event received
```
{
  'version': '0',
  'id': '1967b367-1419-4bf0-87ee-70d64740e942',
  'detail-type': 'Scheduled Event', 
  'source': 'aws.scheduler', 
  'account': '111111111111', 
  'time': '2025-02-17T16:43:00Z', 
  'region': 'eu-west-1', 
  'resources': ['arn:aws:scheduler:eu-west-1:111111111111:schedule/default/example-scheduler'], 
  'detail': '{}'
}
```

And the returned error
```
[ERROR] InvalidModelTypeError: Error: 3 validation errors for EventBridgeModel
detail.dict[str,any]
  Input should be a valid dictionary [type=dict_type, input_value='{}', input_type=str]
    For further information visit https://errors.pydantic.dev/2.9/v/dict_type detail.is-subclass[BaseModel]
  
Input should be a subclass of BaseModel [type=is_subclass_of, input_value='{}', input_type=str]
    For further information visit https://errors.pydantic.dev/2.9/v/is_subclass_of detail.BaseModel
  
Input should be a valid dictionary or instance of BaseModel [type=model_type, input_value='{}', input_type=str]
    For further information visit https://errors.pydantic.dev/2.9/v/model_type. Please ensure the Input model inherits from BaseModel,
and your payload adheres to the specified Input model structure.
Model=<class 'aws_lambda_powertools.utilities.parser.models.event_bridge.EventBridgeModel'>
```


### Code snippet

```python
event = {
  'version': '0',
  'id': '1967b367-1419-4bf0-87ee-70d64740e942',
  'detail-type': 'Scheduled Event', 
  'source': 'aws.scheduler', 
  'account': '111111111111', 
  'time': '2025-02-17T16:43:00Z', 
  'region': 'eu-west-1', 
  'resources': ['arn:aws:scheduler:eu-west-1:111111111111:schedule/default/example-scheduler'], 
  'detail': '{}'
}

parse(event, model=EventBridgeModel)


# Or with an actual event bridge scheduler invoking the lambda
@event_parser(model=EventBridgeModel)
def handler(event, context: LambdaContext):
    print(event)
```

### Possible Solution

- Maybe a pre validator if type is string for `detail` property that json.loads() the value

### Steps to Reproduce

- Create a lambda with a simple handler that parse `EventBridgeModel`
- Create an eventbridge scheduler with a near CRON time.
- Make sure the target role has `lambda:InvokeFunction` policy
- Consult CloudWatch logs

### Powertools for AWS Lambda (Python) version

latest

### AWS Lambda function runtime

3.12

### Packaging format used

Lambda Layers

### Debugging logs

```python

```

--- a/aws_lambda_powertools/utilities/parser/models/event_bridge.py
+++ b/aws_lambda_powertools/utilities/parser/models/event_bridge.py
@@ -1,7 +1,7 @@
 from datetime import datetime
 from typing import List, Optional
 
-from pydantic import BaseModel, ConfigDict, Field
+from pydantic import BaseModel, ConfigDict, Field, field_validator
 
 from aws_lambda_powertools.utilities.parser.types import RawDictOrModel
 
@@ -19,3 +19,9 @@ class EventBridgeModel(BaseModel):
     detail_type: str = Field(..., alias="detail-type")
     detail: RawDictOrModel
     replay_name: Optional[str] = Field(None, alias="replay-name")
+
+    @field_validator("detail", mode="before")
+    def validate_detail(cls, v, fields):
+        # EventBridge Scheduler sends detail field as '{}' string when no payload is present
+        # See: https://github.com/aws-powertools/powertools-lambda-python/issues/6112
+        return {} if fields.data.get("source") == "aws.scheduler" and v == "{}" else v

--- a/tests/events/eventBridgeSchedulerEvent.json
+++ b/tests/events/eventBridgeSchedulerEvent.json
@@ -3,7 +3,7 @@
   "id":"d167b752-343a-4b28-afd6-d4de056319e8",
   "detail-type":"Scheduled Event",
   "source":"aws.scheduler",
-  "account":"533568316194",
+  "account":"123456789012",
   "time":"2025-02-20T16:03:00Z",
   "region":"us-east-1",
   "resources":[

--- a/tests/events/eventBridgeSchedulerEvent.json
+++ b/tests/events/eventBridgeSchedulerEvent.json
@@ -1,0 +1,13 @@
+{
+  "version":"0",
+  "id":"d167b752-343a-4b28-afd6-d4de056319e8",
+  "detail-type":"Scheduled Event",
+  "source":"aws.scheduler",
+  "account":"533568316194",
+  "time":"2025-02-20T16:03:00Z",
+  "region":"us-east-1",
+  "resources":[
+     "arn:aws:scheduler:us-east-1:123456789012:schedule/default/aaaaa"
+  ],
+  "detail":"{}"
+}

--- a/tests/unit/parser/_pydantic/test_eventbridge.py
+++ b/tests/unit/parser/_pydantic/test_eventbridge.py
@@ -1,6 +1,7 @@
 import pytest
 
 from aws_lambda_powertools.utilities.parser import ValidationError, envelopes, parse
+from aws_lambda_powertools.utilities.parser.models import EventBridgeModel
 from tests.functional.utils import load_event
 from tests.unit.parser._pydantic.schemas import (
     MyAdvancedEventbridgeBusiness,
@@ -51,3 +52,10 @@ def test_handle_invalid_event_with_eventbridge_envelope():
     empty_event = {}
     with pytest.raises(ValidationError):
         parse(event=empty_event, model=MyEventbridgeBusiness, envelope=envelopes.EventBridgeEnvelope)
+
+
+def test_handle_eventbridge_scheduler():
+    raw_event = load_event("eventBridgeSchedulerEvent.json")
+    parsed_event: EventBridgeModel = EventBridgeModel(**raw_event)
+
+    assert parsed_event.detail == {}

